core/src/main/java/org/elasticsearch/action/admin/indices/analyze/AnalyzeRequest.java

```diff
@@ -82,35 +136,77 @@ public String analyzer() {
     }
 
     public AnalyzeRequest tokenizer(String tokenizer) {
-        this.tokenizer = tokenizer;
+        if (tokenizer != null) {
+            XContentType type = XContentFactory.xContentType(tokenizer);
```

Comment:
this method should assume that `tokenizer` is a tokenizer name rather than trying to parse it as json